### DISCUSSION TITLE
How should I send image data as File upload together with json field as array values ?

### DISCUSSION THREAD

**Discussion by yichong96** [fastapi/fastapi#4297] [Questions]
I would like to receive image data together with some metadata for the image. I understand that Fast API is able to support multi-part form encoding by specifying `Form` and `File` fields. However, the Form fields support only "key":"value" data. 

Ideally, I would like to receive the metadata as json as one of the fields for the metadata contains array values. 

Is there a workaround to this ? Would encoding the image data as base 64 and serialising the data using pydantic model be the best
choice ?

Thank you for your advice !

**Comment by bluebrown** [comment]
You can technically use multipart/mixed content type. I haven't checked how easy it is to implement with fastapi though.
